contracts/src/7683/T1ERC7683Pull.sol

```diff
@@ -121,31 +106,22 @@ contract T1ERC7683Pull is BasicSwap7683, OwnableUpgradeable, IT1XChainReaderCall
         emit SettlementVerificationRequested(orderId, requestId);
     }
 
-    function onT1XChainReaderResult(bytes32 requestId, uint256 batchIndex, bytes32 newRoot) external onlyXChainRead {
-        requestIdToBatchIndex[requestId] = batchIndex;
-        proofOfReadRoots[batchIndex] = newRoot;
-
-        emit ProofOfReadRootCommitted(batchIndex);
-    }
-
-    // @notice Callback function for cross-chain read results
-    // @param _originDomain The origin domain
-    // @param _sender The sender address
-    // @param requestId The ID of the read request
-    // @param result The result of the read
+    /// @notice Use result of proof of read to handle the order depending on the result
+    /// @param batchIndex The batch index of the read request
+    /// @param requestId The ID of the read request
+    /// @param result The result of the read
+    /// @param position The position of the read request in the merkle tree
+    /// @param proof The proof of the read request
     function handleReadResultWithProof(
+        uint256 batchIndex,
         bytes32 requestId,
+        uint256 position,
         bytes calldata result,
-        uint256 nonce,
         bytes calldata proof
     )
         external
     {
-        bytes32 xChainReadResultHash = keccak256(result);
-        bytes32 leaf = keccak256(abi.encodePacked(xChainReadResultHash, requestId));
-        uint256 batchIndex = requestIdToBatchIndex[requestId];
-
-        if (!WithdrawTrieVerifier.verifyMerkleProof(proofOfReadRoots[batchIndex], leaf, nonce, proof)) {
+        if (!xChainRead.verifyProofOfRead(batchIndex, requestId, position, result, proof)) {
```

Comment:
contrary to my initial suggestion, I think we should pack all fields into a single `bytes` param to allow us maximum flexibility in later reconfiguring the params without affecting any consumers of `xChainRead`